File: TitanEngine/SDK/Samples/Scripts/LUA/Unpackers/Static unpackers/LameCrypt.lua
require 'SDK'

    local SystemKernel2 = alien.load 'kernel32.dll'

-- Variables
    local pFileMap = alien.buffer(alien.sizeof('pointer'))
	pFileMap:set(1,nil,'pointer')
    local pFileHandle = alien.buffer(alien.sizeof('pointer'))
	pFileHandle:set(1,nil,'pointer')
    local pFileMapVA = alien.buffer(alien.sizeof('pointer'))
	pFileMapVA:set(1,nil,'pointer')
    local pLoadedSize = alien.buffer(alien.sizeof('pointer'))
	pLoadedSize:set(1,nil,'pointer')
    local pUnpackedOEP = alien.buffer(alien.sizeof('pointer'))
	pUnpackedOEP:set(1,nil,'pointer')

    SystemKernel2.RtlMoveMemory:types {"pointer","long","long",abi="stdcall"}
    SK_RtlMoveMemory = SystemKernel2.RtlMoveMemory


-- Code

	inFileName = arg[1]
	BackFileName = inFileName .. '.bak'
	print("Unpacker for LameCrypt 1.0\nfrom ReversingLabs Corporation - www.reversinglabs.com\n\n");
	bReturn = TE_CopyFileA(inFileName, BackFileName, TE_FALSE)
	if not(bReturn == 0) then
		bReturn = TE_StaticFileLoad(inFileName, UE_ACCESS_ALL, TE_FALSE, pFileHandle, pLoadedSize, pFileMap, pFileMapVA)
		if not(bReturn == 0) then
			-- Convert pointers to values
			FileMap = pFileMap:get(1,'long')
			FileHandle = pFileHandle:get(1,'long')
			FileMapVA = pFileMapVA:get(1,'long')
			LoadedSize = pLoadedSize:get(1,'long')
			--
			-- Unpack file
			rUnpackedOEP = TE_ConvertVAtoFileOffset(FileMapVA, TE_GetPE32DataFromMappedFile(FileMapVA, nil, UE_OEP) + TE_GetPE32DataFromMappedFile(FileMapVA, NULL, UE_IMAGEBASE), TE_TRUE) + 0x19
			SK_RtlMoveMemory(pUnpackedOEP, rUnpackedOEP, 4)
			UnpackedOEP = pUnpackedOEP:get(1,'long') - TE_GetPE32DataFromMappedFile(FileMapVA, NULL, UE_IMAGEBASE)
			TE_StaticSectionDecrypt(FileMapVA, nil, TE_FALSE, UE_STATIC_DECRYPTOR_XOR, UE_STATIC_KEY_SIZE_1, 0x90)
			TE_SetPE32DataForMappedFile(FileMapVA, nil, UE_OEP, UnpackedOEP)
			TE_StaticFileUnload(inFileName, TE_TRUE, FileHandle, LoadedSize, FileMap, FileMapVA)
			TE_DeleteLastSection(inFileName)
			print("Selected file has been unpacked!\n");
		else
			print("Could not open selected file!")
		end
	else
		print("Could not backup selected file!")
	end

File: TitanEngine/SDK/Samples/Unpackers/x86/C/Static unpackers/LameCrypt/LameCrypt/LameCrypt.cpp
// LameCrypt.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include <windows.h>
#include "SDK\SDK.h"


int main(int argc, char* argv[]){

	HANDLE FileHandle;
	DWORD FileSize;
	HANDLE FileMap;
	ULONG_PTR FileMapVA;
	DWORD UnpackedOEP;
	char szOriginalFile[MAX_PATH] = {};

	lstrcpyA(szOriginalFile, argv[1]);
	lstrcatA(szOriginalFile, ".bak");
	printf("Unpacker for LameCrypt 1.0\r\nfrom ReversingLabs Corporation - www.reversinglabs.com\r\n\r\n");
	if(CopyFileA(argv[1], szOriginalFile, false)){
		if(StaticFileLoad(argv[1], UE_ACCESS_ALL, false, &FileHandle, &FileSize, &FileMap, &FileMapVA)){
			RtlMoveMemory(&UnpackedOEP, (void*)(ConvertVAtoFileOffset(FileMapVA, (ULONG_PTR)GetPE32DataFromMappedFile(FileMapVA, NULL, UE_OEP) + (ULONG_PTR)GetPE32DataFromMappedFile(FileMapVA, NULL, UE_IMAGEBASE), true) + 0x19), 4);
			UnpackedOEP = UnpackedOEP - (DWORD)GetPE32DataFromMappedFile(FileMapVA, NULL, UE_IMAGEBASE);
			StaticSectionDecrypt(FileMapVA, NULL, false, UE_STATIC_DECRYPTOR_XOR, UE_STATIC_KEY_SIZE_1, 0x90);
			SetPE32DataForMappedFile(FileMapVA, NULL, UE_OEP, (ULONG_PTR)UnpackedOEP);
			StaticFileUnload(argv[1], true, FileHandle, FileSize, FileMap, FileMapVA);
			DeleteLastSection(argv[1]);
			printf("Selected file has been unpacked!\r\n");
		}else{
			printf("Could not find selected input file!\r\n");
		}
	}else{
		printf("Could not backup selected input file!\r\n");
	}
	return 0;
}

